File: testdata/bench/kotlin/BenchmarkTest00044.kt
package bench

import io.ktor.server.application.*
import io.ktor.server.request.*
import io.ktor.server.response.*

// Vulnerable: command injection via receive body
suspend fun handler00044(call: ApplicationCall) {
    val body = call.receive<Map<String, String>>()
    val ip = body["ip"]
    val process = Runtime.getRuntime().exec("nslookup $ip")
    val output = process.inputStream.bufferedReader().readText()
    call.respondText(output)
}

File: testdata/bench/go/BenchmarkTest00044.go
package bench

import (
	"io"
	"net/http"
	"os"
)

// Path traversal via os.Create
func Handler00044(w http.ResponseWriter, r *http.Request) {
	dest := r.FormValue("dest")
	f, _ := os.Create(dest)
	defer f.Close()
	io.Copy(f, r.Body)
}
